spec/rspec/rails/example/controller_example_group_spec.rb

```diff
@@ -19,6 +19,23 @@ def group_for(klass)
       expect(group.included_modules).to include(RSpec::Rails::Matchers::RoutingMatchers)
     end
 
+    it "handles methods invoked via `method_missing` that use keywords" do
+      group =
+        RSpec::Core::ExampleGroup.describe ApplicationController do
+          def a_method(value:); value; end
+          def method_missing(_name, *_args, **kwargs, &_block); a_method(**kwargs); end
```

Comment:
WDYT of writing this like this instead:
```diff
- def method_missing(_name, *_args, **kwargs, &_block); a_method(**kwargs); end
+ def method_missing(_name, *args, &_block); a_method(*args); end
+ ruby2_keywords :method_missing if respond_to?(:ruby2_keywords, true)
```

The error message (without the fix) is:
```
     ArgumentError:
       wrong number of arguments (given 1, expected 0; required keyword: value)
 ```
instead of:
```
     ArgumentError:
       missing keyword: :value
```

Reply:
Its purposefully demonstrating the mis match between keyword arguments, on Ruby 3.0.2 on my machine the error is:

```
Failure/Error: def a_method(value:); value; end

     ArgumentError:
       missing keyword: :value
```

Reply:
Agree. Tried a couple of different cases (dafault kwargs, do passing an argument in addition to kwargs), and with your approach, the failure message is consistent.